### PR TITLE
save_diagnostic_files

### PR DESCRIPTION
**Pull request by RomanPudashkin**
Resolves: https://github.com/musescore/MuseScore/issues/15830
Resolves: https://github.com/musescore/MuseScore/issues/15145

### CODE REVIEW

File: src/appshell/qml/DevTools/DevToolsPage.qml [hunk 1]
```diff
@@ -49,10 +48,6 @@ DockPage {
         case "gallery": root.central = galleryComp; break
         case "interactive": root.central = interactiveComp; break
         case "crashhandler": root.central = crashhandlerComp; break
-        case "audio": root.central = audioComp; break
-        case "synth": root.central = synthSettingsComp; break
-        case "midiports": root.central = midiPortsComp; break
-        case "vst": root.central = vstComponent; break
         case "mpe": root.central = mpeComponent; break
         case "autobot": root.central = autobotComp; break
```
Comment (igorkorsukov) [inline src/appshell/qml/DevTools/DevToolsPage.qml:52]: can be removed too

File: src/diagnostics/diagnosticsmodule.cpp [hunk 2]
```diff
@@ -72,6 +72,7 @@ void DiagnosticsModule::registerExports()
     ioc()->registerExport<IDiagnosticsPathsRegister>(moduleName(), new DiagnosticsPathsRegister());
     ioc()->registerExport<IEngravingElementsProvider>(moduleName(), new EngravingElementsProvider());
     ioc()->registerExport<IEngravingDrawProvider>(moduleName(), new EngravingDrawProvider());
+    ioc()->registerExport<IDiagnosticsConfiguration>(moduleName(), s_configuration);
```
Comment (igorkorsukov) [inline src/diagnostics/diagnosticsmodule.cpp:75]: It is better to change to:

```
static std::shared_ptr<DiagnosticsConfiguration> s_configuration = {};
static std::shared_ptr<DiagnosticsActionsController> s_actionsController = {};

...
void DiagnosticsModule::registerExports()
{
    s_configuration = std::make_shared<DiagnosticsConfiguration>();
    s_actionsController = std::make_shared<DiagnosticsActionsController>();
```

File: src/diagnostics/internal/diagnosticsactionscontroller.cpp [hunk 3]
```diff
@@ -52,3 +62,55 @@ void DiagnosticsActionsController::openUri(const mu::UriQuery& uri, bool isSingl
...
+
+    qApp->restoreOverrideCursor();
+
+    if (!ret) {
+        LOGE() << ret.toString();
+        return;
+    }
+
+    interactive()->revealInFileBrowser(path);
+}
```
Comment (igorkorsukov) [inline src/diagnostics/internal/diagnosticsactionscontroller.cpp:106]: I would prefer that the implementation of this function was not in the controller. Maybe it's better to move it to DiagnosticFilesWriter (and make it a regular service with access through the interface)

**Review by igorkorsukov — APPROVED** [review]

File: src/diagnostics/internal/diagnosticsactionscontroller.cpp (hunk 3 above)
Comment (RomanPudashkin) [reply src/diagnostics/internal/diagnosticsactionscontroller.cpp:67]: moved into a separate class

**Comment by RomanPudashkin** [comment]
@DmitryArefiev stable builds containing these changes for testing purposes:

win: https://github.com/RomanPudashkin/MuseScore/actions/runs/3931921766
macOS: https://github.com/RomanPudashkin/MuseScore/actions/runs/3931923262
linux: https://github.com/RomanPudashkin/MuseScore/actions/runs/3931924220

**Comment by DmitryArefiev** [comment]
Tested #15830 #15145 on Win10, Mac12, LinuxMint 22.04 - **PASS**

**Pull request merged**